server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -724,4 +756,86 @@ public int getConsumerEntitlementCount(Owner owner, ConsumerType type) {
             .setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY)
             .list();
     }
+    /**
+     * InExpressionIgnoringCase
+     *
+     * Allows Criterion for case insensitive comparison to list of values
+     *
+     * @author wpoteat
+     */
+    public class InExpressionIgnoringCase implements Criterion {
+
+        private final String propertyName;
+        private final Object[] values;
+
+        public InExpressionIgnoringCase(final String propertyName, final Object[] values) {
+            this.propertyName = propertyName;
+            this.values = values;
+        }
+
+        public String toSqlString(final Criteria criteria, final CriteriaQuery criteriaQuery)
+            throws HibernateException {
+            final String[] columns = criteriaQuery.findColumns(this.propertyName, criteria);
+            final String[] wrappedLowerColumns = wrapLower(columns);
+            if (criteriaQuery.getFactory().getDialect().supportsRowValueConstructorSyntaxInInList() ||
+                    columns.length <= 1) {
+
+                String singleValueParam = StringHelper.repeat("lower(?), ", columns.length - 1) + "lower(?)";
+                if (columns.length > 1) {
+                    singleValueParam = '(' + singleValueParam + ')';
+                }
+                final String params = this.values.length > 0 ? StringHelper.repeat(singleValueParam + ", ",
+                        this.values.length - 1) + singleValueParam : "";
+                String cols = StringHelper.join(", ", wrappedLowerColumns);
+                if (columns.length > 1) {
+                    cols = '(' + cols + ')';
+                }
+                return cols + " in (" + params + ')';
+            }
+            else {
+                String cols = " ( " + StringHelper.join(" = lower(?) and ",
+                        wrappedLowerColumns) + "= lower(?) ) ";
+                cols = this.values.length > 0 ? StringHelper.repeat(cols + "or ",
+                        this.values.length - 1) + cols : "";
+                cols = " ( " + cols + " ) ";
+                return cols;
+            }
+        }
+
+        public TypedValue[] getTypedValues(final Criteria criteria, final CriteriaQuery criteriaQuery)
+            throws HibernateException {
+            final ArrayList<TypedValue> list = new ArrayList<TypedValue>();
```

Comment:
Good call on the use of ArrayList here -- converts to an array really quickly. You may want to try giving it a default size, though, to avoid worrying about several reallocations and copies as the contents grow.

Side note: Why are all the variables declared as final here?